packages/core/src/UserState.ts

```diff
@@ -172,33 +192,26 @@ export default class UserState {
         return foundLeaf.index
     }
 
-    getEpochKeys(_epoch?: bigint | number, nonce?: number) {
-        const epoch = _epoch ?? this.sync.calcCurrentEpoch()
-        if (
-            typeof nonce === 'number' &&
-            nonce >= this.sync.settings.numEpochKeyNoncePerEpoch
-        ) {
-            throw new Error(
-                `getEpochKeys nonce ${nonce} exceeds max nonce value ${this.sync.settings.numEpochKeyNoncePerEpoch}`
-            )
-        }
+    getEpochKeys(
+        _epoch?: bigint | number,
+        nonce?: number,
+        attesterId: bigint | string = this.sync.attesterId
```

Comment:
Not sure what will happen is `sync` is `undefined`.

Reply:
Ah it comes with
`@unirep/core:Synchronizer: attester ID 0 is not synchronized` first 
instead of 
`@unirep/core:UserState: no synchronizer is set`

Reply:
[63ddb2d](https://github.com/Unirep/Unirep/pull/367/commits/63ddb2df7de19cd86a2e90a24b830dc20557bcbd)
set a better error message here
if there is at least one attester synced, it takes the first one.
and the default one can be set with `setAttesterId()`